### PR TITLE
feat: dynamic CLI title and terminal utils refactor

### PR DESCRIPTION
**Pull request by edmarpaulino**
This pull request refactors and improves terminal UI utilities in the CLI package. The main changes include centralizing terminal dimension and text truncation logic into a new `utils` module, updating all usages to the new utilities, and enhancing test coverage for these functions. Additionally, the ASCII art logo now adapts to terminal width.

**UI Utilities Refactor and Improvements:**

* Introduced a new `ui/utils.ts` module containing `getTerminalWidth`, `getTerminalHeight`, `truncateText`, and terminal defaults for width and height. These functions standardize how terminal dimensions and text truncation are handled across the CLI.
* Updated all references to the old `truncate` and terminal dimension logic in `input.ts`, `install.ts`, and `utils.ts` to use the new `truncateText`, `getTerminalWidth`, and `getTerminalHeight` utilities. Removed the now-obsolete `truncate` function from `formatting.ts`. [[1]](diffhunk://#diff-cb1a007e0e4820237308d398e754e743fc8863ffb0420c5b26a59c0d3c311062L8) [[2]](diffhunk://#diff-cb1a007e0e4820237308d398e754e743fc8863ffb0420c5b26a59c0d3c311062R18) [[3]](diffhunk://#diff-cb1a007e0e4820237308d398e754e743fc8863ffb0420c5b26a59c0d3c311062L353-R353) [[4]](diffhunk://#diff-fb7df92176cb49ec27e3dd8b96281d4f24034948b18e9b3674db8e3eb971a7b1L7-R8) [[5]](diffhunk://#diff-fb7df92176cb49ec27e3dd8b96281d4f24034948b18e9b3674db8e3eb971a7b1L29-R29) [[6]](diffhunk://#diff-44b131669ddb36088bc9ec58834ced4b2f4de9b0a5d7bc4076c332f0a97e4fa2L1-L3) [[7]](diffhunk://#diff-fc066b1814f7a8d1728fc20d857148672343d3ec5cd4c5d1d5c5a96a15854cdaR13) [[8]](diffhunk://#diff-fc066b1814f7a8d1728fc20d857148672343d3ec5cd4c5d1d5c5a96a15854cdaL22-L36)

**Testing Enhancements:**

* Added comprehensive unit tests for `truncateText`, `getTerminalWidth`, and `getTerminalHeight` in `ui/utils.spec.ts`, ensuring correct handling of edge cases and terminal environment variations.

**UI/UX Improvements:**

* Improved the ASCII art logo in `screen.ts` to dynamically adjust between the full "Tech Leads Club" and a shortened "TLC" version based on the current terminal width, making the CLI more visually adaptable.

### CODE REVIEW

File: packages/cli/src/ui/utils.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+export function getTerminalWidth(defaultWidth = TERMINAL_DEFAULTS.WIDTH): number {
+  return process.stdout.columns ?? defaultWidth
+}
+
+export function getTerminalHeight(defaultHeight = TERMINAL_DEFAULTS.HEIGHT): number {
+  return process.stdout.rows ?? defaultHeight
+}
+
+export function truncateText(text: string, maxWidth: number): string {
+  if (text.length <= maxWidth) return text
```
[the remark this answers is not in the document]
Comment (edmarpaulino) [reply packages/cli/src/ui/utils.ts]: Suggestion applied.

File: packages/cli/src/ui/screen.ts [hunk 2]
```diff
@@ -2,9 +2,12 @@ import figlet from 'figlet'
 import pc from 'picocolors'
 
 import { crystalGradient } from './styles'
+import { getTerminalWidth } from './utils'
 
 export function generateLogo(): string {
-  const asciiArt = figlet.textSync('Tech Leads Club', { font: 'Larry 3D', horizontalLayout: 'default' })
+  const width = getTerminalWidth()
+  const title = width < 112 ? 'TLC' : 'Tech Leads Club'
+  const asciiArt = figlet.textSync(title, { font: 'Larry 3D', horizontalLayout: 'default' })
```
[the remark this answers is not in the document]
Comment (edmarpaulino) [reply packages/cli/src/ui/screen.ts]: Fixed.

File: packages/cli/src/ui/utils.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  return text.slice(0, Math.max(0, maxWidth - 3)) + '...'
+}
```
[the remark this answers is not in the document]
Comment (edmarpaulino) [reply packages/cli/src/ui/utils.ts]: Suggestion applied.

File: packages/cli/src/ui/utils.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+export const TERMINAL_DEFAULTS = {
+  WIDTH: 80,
+  HEIGHT: 20,
+}
```
[the remark this answers is not in the document]
Comment (edmarpaulino) [reply packages/cli/src/ui/utils.ts]: Suggestion applied.

File: packages/cli/src/ui/screen.ts [hunk 5]
```diff
@@ -2,9 +2,18 @@ import figlet from 'figlet'
...
-  const asciiArt = figlet.textSync('Tech Leads Club', { font: 'Larry 3D', horizontalLayout: 'default' })
+  const width = getTerminalWidth()
+  const fullTitle = 'Tech Leads Club'
+  const fullArt = figlet.textSync(fullTitle, { font: 'Larry 3D', horizontalLayout: 'default' })
+  const artWidth = fullArt.split('\n').reduce((max, line) => Math.max(max, line.length), 0)
+
+  const title = width < artWidth + 2 ? 'TLC' : fullTitle
+  const asciiArt =
+    title === fullTitle ? fullArt : figlet.textSync(title, { font: 'Larry 3D', horizontalLayout: 'default' })
+
```
[the remark this answers is not in the document]
Comment (edmarpaulino) [reply packages/cli/src/ui/screen.ts:18]: Suggestion applied.

[the remark this answers is not in the document]
Comment (edmarpaulino) [reply packages/cli/src/ui/screen.ts:18]: PR description updated.

File: packages/cli/src/ui/utils.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+export function getTerminalHeight(defaultHeight = TERMINAL_DEFAULTS.HEIGHT): number {
+  return process.stdout.rows ?? defaultHeight
+}
+
+export function truncateText(text: string, maxLength: number): string {
+  if (maxLength <= 0) return ''
+  if (text.length <= maxLength) return text
+  if (maxLength < 3) return '.'.repeat(maxLength)
+  return text.slice(0, Math.max(0, maxLength - 3)) + '...'
+}
```
[the remark this answers is not in the document]
Comment (edmarpaulino) [reply packages/cli/src/ui/utils.ts:19]: Suggestion applied.

File: packages/cli/src/__tests__/ui/utils.spec.ts [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        configurable: true,
+      })
+    }
+
+    afterEach(() => {
+      Object.defineProperty(process, 'stdout', {
+        value: originalStdout,
+        writable: true,
+        configurable: true,
+      })
```
[the remark this answers is not in the document]
Comment (edmarpaulino) [reply packages/cli/src/__tests__/ui/utils.spec.ts]: Fixed.

File: packages/cli/src/__tests__/ui/utils.spec.ts [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+  describe('Terminal Dimensions', () => {
+    const originalColumns = process.stdout.columns
+    const originalRows = process.stdout.rows
+
+    const mockStdout = (columns?: number, rows?: number) => {
+      Object.defineProperty(process.stdout, 'columns', {
+        value: columns,
+        configurable: true,
+      })
```
[the remark this answers is not in the document]
Comment (edmarpaulino) [reply packages/cli/src/__tests__/ui/utils.spec.ts:40]: Fixed.
